projects/go-lib/src/lib/components/go-side-nav/go-side-nav/go-side-nav.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Router, NavigationEnd } from '@angular/router';
+import { filter } from 'rxjs/operators';
+import { NavGroup } from '../nav-group.model';
+import { NavItem } from '../nav-item.model';
+import { GoSideNavService } from './go-side-nav.service';
+
+@Component({
+  selector: 'go-side-nav',
+  templateUrl: './go-side-nav.component.html',
+  styleUrls: ['./go-side-nav.component.scss']
+})
+export class GoSideNavComponent implements OnInit {
+  @Input() menuItems: Array<NavGroup | NavItem>;
+
+  constructor (private router: Router, public navService: GoSideNavService) { }
+
+  ngOnInit() {
+    this.router.events
+      .pipe(
+        filter(event => event instanceof NavigationEnd)
+      ).subscribe((event: NavigationEnd) => {
+        this.menuItems.forEach(item => {
+          (item as NavGroup).expanded = this.setExpanded(item, event.url);
+        });
+    });
+  }
+
+  closeNavs(item) {
+    this.menuItems.forEach(group => {
+      (group as NavGroup).expanded = this.openAccordion((group as NavGroup), item);
+    });
+  }
+
+  //#region Private Methods
+
+  private setExpanded(route: NavGroup | NavItem, url: string) {
+    if ((route as NavGroup).subRoutes) {
+      return this.navGroupExpansion(route as NavGroup, url);
+    } else {
+      return url.includes((route as NavItem).route);
+    }
+  }
+
+  private navGroupExpansion(route: NavGroup, url: string) {
+    route.expanded = route.subRoutes.some(subRoute => {
+      return this.setExpanded(subRoute, url);
+    });
+    return route.expanded;
+  }
+
+  private openAccordion(group: NavGroup, item: NavGroup): boolean {
```

Comment:
This whole function seems to be very complex. Should we leave a comment above or somewhere in our documentation what this actually does?

I tried to come up with a better solution but couldn't make it dynamically. Maybe @grahamhency has another suggestion for this?